Cargo.toml

```
@@ -9,7 +9,7 @@ version = "0.9.1"
 
 [dependencies]
 quickcheck = { version = "0.6", optional = true }
-serde = { version = "1.0", optional = true }
+serde = { version = "1.0.25", optional = true }
```

Do we depend on features introduced in 1.0.25? I think it's good practice to keep these dependency specifications as general as possible, to be compatible to binaries locking their dependencies to earlier versions.

For example adding `ascii` to crate `example` shouldn't neccessarily force a update of serde even though `example` already uses an earlier version of serde before 1.0.25.

```suggestion
serde = { version = "1.0", optional = true }
```

---

Yes, with serde `version = "=1.0.24"` and `version = "=1.0.0"`, I get
```
error[E0407]: method `deserialize_in_place` is not a member of trait `Deserialize`
   --> src/serialization/ascii_string.rs:100:5
    |
100 | /     fn deserialize_in_place<D>(deserializer: D, place: &mut Self) -> Result<(), D::Error>
101 | |     where
102 | |         D: Deserializer<'de>,
103 | |     {
104 | |         deserializer.deserialize_string(AsciiStringInPlaceVisitor(place))
105 | |     }
    | |_____^ not a member of trait `Deserialize`
```
, so `Deserialize::deserialize_in_place()` must have been added in serde 1.0.25.

I've also tested that the minimum version won't affect other crates unless the feature is enabled.

---

Thanks for the clarification. Obviously that was already true before your PR, so it's a nice fix.